src/query/server/server_test.go

```diff
@@ -67,9 +67,18 @@ clusters:
         type: unaggregated
         retention: 48h
 
-workerPoolCount: 20
-decompressWorkerPoolSize: 20
-writeWorkerPoolSize: 20
+
+readWorkerPoolPolicy:
+  grow: true
+  size: 100
```

Comment:
size has to be at least as large as shards: https://github.com/m3db/m3x/blob/master/sync/pooled_worker_pool.go#L54

Also this configuration seems a little weird, you'll have a pool size of 100 by default that will expand on demand, but each goroutine will have a 30% probability of killing itself.  I'd probably increase the size (should at least be the same as shards), maybe reduce shards a bit, and then decrease the kill probability to something like 0.01

Reply:
Also note that with `grow: true` parallelism is unbounded

Reply:
Yeah, these are just random numbers since it's more just a test to ensure the config gets read correctly